Clean up a policy concept page: resource-quotas (#48944)
kubernetes/website: content/en/docs/concepts/policy/resource-quotas.md

As overcommit is not allowed for extended resources, it makes no sense to specify both `requests`
and `limits` for the same extended resource in a quota. So for extended resources, only quota items

> with prefix `requests.` are allowed for now.

Reviewer: (nit)
```suggestion
with prefix `requests.` are allowed.
```

## Storage Resource Quota

You can limit the total sum of [storage resources](/docs/concepts/storage/persistent-volumes/)
that can be requested in a given namespace.

In addition, you can limit consumption of storage resources based on associated storage-class.

| Resource Name | Description |
| ------------- | ----------- |
| `requests.storage` | Across all persistent volume claims, the sum of storage requests cannot exceed this value. |
| `persistentvolumeclaims` | The total number of [PersistentVolumeClaims](/docs/concepts/storage/persistent-volumes/#persistentvolumeclaims) that can exist in the namespace. |
| `<storage-class-name>.storageclass.storage.k8s.io/requests.storage` | Across all persistent volume claims associated with the `<storage-class-name>`, the sum of storage requests cannot exceed this value. |
| `<storage-class-name>.storageclass.storage.k8s.io/persistentvolumeclaims` | Across all persistent volume claims associated with the `<storage-class-name>`, the total number of [persistent volume claims](/docs/concepts/storage/persistent-volumes/#persistentvolumeclaims) that can exist in the namespace. |

> For example, if an operator wants to quota storage with `gold` storage class separate from
> `bronze` storage class, the operator can define a quota as follows:

Reviewer: ```suggestion
For example, if you to quota storage with `gold` StorageClass separate from
a `bronze` StorageClass, you can define a quota as follows:
```

to align more with our style guide.

* `gold.storageclass.storage.k8s.io/requests.storage: 500Gi`
* `bronze.storageclass.storage.k8s.io/requests.storage: 100Gi`

> In release 1.8, quota support for local ephemeral storage is added as an alpha feature:

Reviewer: Not a pure cleanup, but this is a very stale statement.
